subprojects/tooling-api-builders/build.gradle.kts

```diff
@@ -25,6 +25,7 @@ dependencies {
     implementation(project(":testing-base"))
     implementation(project(":testing-jvm"))
     implementation(project(":tooling-api"))
+    implementation(project(":toolchains-jvm"))
```

Comment:
What code here requires classes from `toolchains-jvm`? I can't find any files in this project which reference any of the toolchains classes

Reply:
Can we change this to a `runtimeOnly` dependency and add a `because`?

Reply:
Or, perhaps there is a missing transitive dependency somewhere

Reply:
So we can make this `testCompileOnly`.  I'll switch it to that.

Reply:
This still seems a little fishy. I looked into it a bit more and it seems like this is required since `TestExecutionBuildTaskSchedulerTest` is `Mock`ing `Test`, which for some reason requires `JavaLauncher` on the classpath. Really, we should be defining `toolchains-jvm` as an `api` dependency for the project where `Test` lives. However, since we do not use `api` dependencies really, I don't think we should start here. 

I think what we have here is mostly fine, though I think its important to explain why we have dependencies declared when they're non-`implementation` dependencies. Can we either:
1. Change this back to `testImplementation` 
2. Add a `because`

Also, we should move this declaration down to other `test*` dependency declarations. We should not be co-locating `implementation` and `testImplementation` dependencies next to each other in the buildscript, for consistency.